app/models/repository.rb

```diff
@@ -1,3 +1,7 @@
 class Repository < KalibroClient::Entities::Processor::Repository
   include KalibroRecord
+
+  def self.latest(count=1)
+    all.sort { |one, another| another.id <=> one.id }.first(count)
```

Comment:
It seems to be way too much effort just for this. I'm fine leaving it as
is, even though I think it's not nearly the best code we could write
considering the *balance* between readability and efficiency. But I think
it's important to understand well how the cache will work before using it
as a replacement for even *thinking* about performance. As usual, "no
silver bullet" is relevant here.

Also, we're introducing the recent things listings as an UX improvement.
How can we know if it will be useful if we don't even know how often it
will be updated?

Em qui, 14 de abr de 2016 12:33, Diego de Araújo Martinez Camarinha <
notifications@github.com> escreveu:

> In app/models/repository.rb
> <https://github.com/mezuro/prezento/pull/337#discussion_r59739014>:
>
> > @@ -1,3 +1,7 @@
> >  class Repository < KalibroClient::Entities::Processor::Repository
> >    include KalibroRecord
> > +
> > +  def self.latest(count=1)
> > +    all.sort { |one, another| another.id <=> one.id }.first(count)
>
> Maybe we can measure the time it takes on both cases. If the difference is
> really of two or three orders of magnitude, I think we should do it even if
> readability gets worse.
>
> The reason for that is we do not want that our pages take more than a few
> hundreds of milliseconds to load. Specially our home page, even on a few
> cases (when cache needs to be updated). There's been a lot of research
> concluding that users will not return to your site if the pages take too
> long to load (usually that means at most 1 second).
>
> —
> You are receiving this because you are on a team that was mentioned.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/mezuro/prezento/pull/337/files/610763c4f6a22b9db9bcf044a0fa990a50ded422#r59739014>
>